### ISSUE TITLE
video_player initialize Exception

### ISSUE THREAD

**Issue by larrykaren** [flutter/flutter#31469]
**This only happens on IOS，
this example run on iPhone X**
## example:
```java 
class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  VideoPlayerController _controller;
  @override
  void initState() {
    super.initState();
    _controller = VideoPlayerController.network('http://www.sample-videos.com/video123/mp4/720/big_buck_bunny_720p_20mb.mp4')
      ..initialize();

  }
  @override
  Widget build(BuildContext context) {
    return Center(
      child: VideoPlayer(_controller),
    );
  }
}
```
## log:
```java
TX and TY are 0. Rotation: 0. Natural width,height: 544.000000, 960.000000
[VERBOSE-2:ui_dart_state.cc(148)] Unhandled Exception: Bad state: Future already completed
#0      _AsyncCompleter.complete (dart:async/future_impl.dart:39:31)
#1      VideoPlayerController.initialize.eventListener (package:video_player/video_player.dart:239:33)
#2      _rootRunUnary (dart:async/zone.dart:1132:38)
#3      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#4      _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#5      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#6      _DelayedData.perform (dart:async/stream_impl.dart:591:14)
#7      _StreamImplEvents.handleNext (dart:async/stream_impl.dart:707:11)
#8      _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:667:7)
#9      _rootRun (dart:async/zone.dart:1120:38)
#10     _CustomZone.run (dart:async/zone.dart:1021:19)
#11     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#12     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#13     _rootRun (dart:async/zone.dart:1124:13)
#14     _CustomZone.run (dart:async/zone.dart:1021:19)
#15     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#16     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#17     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#18     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
[VERBOSE-2:ui_dart_state.cc(148)] Unhandled Exception: Bad state: Future already completed
#0      _AsyncCompleter.complete (dart:async/future_impl.dart:39:31)
#1      VideoPlayerController.initialize.eventListener (package:video_player/video_player.dart:239:33)
#2      _rootRunUnary (dart:async/zone.dart:1132:38)
#3      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#4      _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#5      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#6      _DelayedData.perform (dart:async/stream_impl.dart:591:14)
#7      _StreamImplEvents.handleNext (dart:async/stream_impl.dart:707:11)
#8      _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:667:7)
#9      _rootRun (dart:async/zone.dart:1120:38)
#10     _CustomZone.run (dart:async/zone.dart:1021:19)
#11     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#12     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#13     _rootRun (dart:async/zone.dart:1124:13)
#14     _CustomZone.run (dart:async/zone.dart:1021:19)
#15     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#16     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#17     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#18     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
[VERBOSE-2:ui_dart_state.cc(148)] Unhandled Exception: Bad state: Future already completed
#0      _AsyncCompleter.complete (dart:async/future_impl.dart:39:31)
#1      VideoPlayerController.initialize.eventListener (package:video_player/video_player.dart:239:33)
#2      _rootRunUnary (dart:async/zone.dart:1132:38)
#3      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#4      _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#5      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#6      _DelayedData.perform (dart:async/stream_impl.dart:591:14)
#7      _StreamImplEvents.handleNext (dart:async/stream_impl.dart:707:11)
#8      _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:667:7)
#9      _rootRun (dart:async/zone.dart:1120:38)
#10     _CustomZone.run (dart:async/zone.dart:1021:19)
#11     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#12     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#13     _rootRun (dart:async/zone.dart:1124:13)
#14     _CustomZone.run (dart:async/zone.dart:1021:19)
#15     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#16     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#17     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#18     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
[VERBOSE-2:ui_dart_state.cc(148)] Unhandled Exception: Bad state: Future already completed
#0      _AsyncCompleter.complete (dart:async/future_impl.dart:39:31)
#1      VideoPlayerController.initialize.eventListener (package:video_player/video_player.dart:239:33)
#2      _rootRunUnary (dart:async/zone.dart:1132:38)
#3      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#4      _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#5      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#6      _DelayedData.perform (dart:async/stream_impl.dart:591:14)
#7      _StreamImplEvents.handleNext (dart:async/stream_impl.dart:707:11)
#8      _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:667:7)
#9      _rootRun (dart:async/zone.dart:1120:38)
#10     _CustomZone.run (dart:async/zone.dart:1021:19)
#11     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#12     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#13     _rootRun (dart:async/zone.dart:1124:13)
#14     _CustomZone.run (dart:async/zone.dart:1021:19)
#15     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#16     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#17     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#18     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
[VERBOSE-2:ui_dart_state.cc(148)] Unhandled Exception: Bad state: Future already completed
#0      _AsyncCompleter.complete (dart:async/future_impl.dart:39:31)
#1      VideoPlayerController.initialize.eventListener (package:video_player/video_player.dart:239:33)
#2      _rootRunUnary (dart:async/zone.dart:1132:38)
#3      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#4      _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#5      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#6      _DelayedData.perform (dart:async/stream_impl.dart:591:14)
#7      _StreamImplEvents.handleNext (dart:async/stream_impl.dart:707:11)
#8      _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:667:7)
#9      _rootRun (dart:async/zone.dart:1120:38)
#10     _CustomZone.run (dart:async/zone.dart:1021:19)
#11     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#12     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#13     _rootRun (dart:async/zone.dart:1124:13)
#14     _CustomZone.run (dart:async/zone.dart:1021:19)
#15     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#16     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#17     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#18     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
TX and TY are 0. Rotation: 0. Natural width,height: 1280.000000, 720.000000
[VERBOSE-2:ui_dart_state.cc(148)] Unhandled Exception: Bad state: Future already completed
#0      _AsyncCompleter.complete (dart:async/future_impl.dart:39:31)
#1      VideoPlayerController.initialize.eventListener (package:video_player/video_player.dart:239:33)
#2      _rootRunUnary (dart:async/zone.dart:1132:38)
#3      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#4      _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#5      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#6      _DelayedData.perform (dart:async/stream_impl.dart:591:14)
#7      _StreamImplEvents.handleNext (dart:async/stream_impl.dart:707:11)
#8      _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:667:7)
#9      _rootRun (dart:async/zone.dart:1120:38)
#10     _CustomZone.run (dart:async/zone.dart:1021:19)
#11     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#12     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#13     _rootRun (dart:async/zone.dart:1124:13)
#14     _CustomZone.run (dart:async/zone.dart:1021:19)
#15     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#16     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#17     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#18     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
[VERBOSE-2:ui_dart_state.cc(148)] Unhandled Exception: Bad state: Future already completed
#0      _AsyncCompleter.complete (dart:async/future_impl.dart:39:31)
#1      VideoPlayerController.initialize.eventListener (package:video_player/video_player.dart:239:33)
#2      _rootRunUnary (dart:async/zone.dart:1132:38)
#3      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#4      _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#5      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#6      _DelayedData.perform (dart:async/stream_impl.dart:591:14)
#7      _StreamImplEvents.handleNext (dart:async/stream_impl.dart:707:11)
#8      _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:667:7)
#9      _rootRun (dart:async/zone.dart:1120:38)
#10     _CustomZone.run (dart:async/zone.dart:1021:19)
#11     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#12     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#13     _rootRun (dart:async/zone.dart:1124:13)
#14     _CustomZone.run (dart:async/zone.dart:1021:19)
#15     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#16     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#17     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#18     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
```

**Comment by cbenhagen** [comment]
duplicate of https://github.com/flutter/flutter/issues/19197

**Comment by iskakaushik** [comment]
This should be addressed in version: `0.10.0+8` by this PR: https://github.com/flutter/plugins/pull/1552/ please re-open this if it's still an issue. Thanks!
